### PR TITLE
feat: execute_call

### PR DESCRIPTION
**Pull request by obatirou**
Time spent on this PR:

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Resolves #<Issue number>

## What is the new behavior?

-
-
-

### CODE REVIEW

File: cairo/protocol_handler/src/protocol_handler.cairo [hunk 1]
```diff
@@ -83,8 +105,9 @@ pub mod ProtocolHandler {
 
     // Access controls roles
     const SECURITY_COUNCIL_ROLE: felt252 = selector!("SECURITY_COUNCIL_ROLE");
-    const GUARDIAN_ROLE: felt252 = selector!("GUARDIAN_ROLE");
     const OPERATOR_ROLE: felt252 = selector!("OPERATOR_ROLE");
+    const GUARDIAN_ROLE: felt252 = selector!("GUARDIAN_ROLE");
+    const GAS_PRICE_ADMIN_ROLE: felt252 = selector!("GAS_PRICE_ADMIN_ROLE");
```
Comment (enitrat) [inline cairo/protocol_handler/src/protocol_handler.cairo:110]: actually wondering, why selector! and not just `'GAS_PRICE_ADMIN_ROLE'`

File: cairo/protocol_handler/tests/test_protocol_handler.cairo [hunk 2]
```diff
@@ -401,3 +410,183 @@ fn test_protocol_handler_unpause_should_pass_after_delay() {
...
+    start_cheat_caller_address(protocol_handler.contract_address, operator_mock());
+
+    // Get SafeDispatcher of protocolHandler
+    let safe_dispatcher = IProtocolHandlerSafeDispatcher {
+        contract_address: protocol_handler.contract_address
+    };
+
+    for selector in unauthoried_selectors
+        .span() {
+            // Mock the call to Kakarot upgrade function
```
Comment (enitrat) [inline cairo/protocol_handler/tests/test_protocol_handler.cairo]:
```suggestion
            // Mock the call to the Kakarot entrypoint
```

File: cairo/protocol_handler/tests/test_protocol_handler.cairo [hunk 3]
```diff
@@ -401,3 +410,183 @@ fn test_protocol_handler_unpause_should_pass_after_delay() {
...
+
+            // Call the protocol handler execute_call
+            #[feature("safe_dispatcher")]
+            match safe_dispatcher.execute_call(call) {
+                Result::Ok(_) => panic!("Entrypoint did not panic"),
+                Result::Err(panic_data) => {
+                    assert(*panic_data.at(0) == 'UNAUTHORIZED_SELECTOR', *panic_data.at(0));
+                }
+            };
+        }
```
Comment (enitrat) [inline cairo/protocol_handler/tests/test_protocol_handler.cairo:565]: add an `assert_called` to ensure the Kakarot function has been called

File: cairo/protocol_handler/tests/test_protocol_handler.cairo [hunk 4]
```diff
@@ -401,3 +410,183 @@ fn test_protocol_handler_unpause_should_pass_after_delay() {
...
+        selector!("set_authorized_pre_eip155_tx"),
+        selector!("set_l1_messaging_contract_address"),
+    ];
+
+    // Change caller to operator
+    start_cheat_caller_address(protocol_handler.contract_address, operator_mock());
+
+    for selector in authorized_selectors
+        .span() {
+            // Mock the call to Kakarot upgrade function
```
Comment (enitrat) [inline cairo/protocol_handler/tests/test_protocol_handler.cairo]:
```suggestion
            // Mock the call to Kakarot entrypoint
```

File: cairo/protocol_handler/src/protocol_handler.cairo (hunk 1 above)
Comment (obatirou) [reply cairo/protocol_handler/src/protocol_handler.cairo:110]: Both could work but honestly followed OZ convention in documentation

File: cairo/protocol_handler/tests/test_protocol_handler.cairo (hunk 3 above)
Comment (obatirou) [reply cairo/protocol_handler/tests/test_protocol_handler.cairo:565]: I do not think we need to: this is a failing test where the error is raised even before the Kakarot function is called.

Comment (enitrat) [reply cairo/protocol_handler/tests/test_protocol_handler.cairo:565]: yes i realised this later but forgot to delete, oops 👻

**Review by enitrat — APPROVED** [review]

**Pull request merged**